src/js/cf-expandables.js

```diff
@@ -5,110 +5,238 @@
+    },
+
+    // Elements that will be cached.
+    ui: {
+      content: '.expandable_content',
+      target:  '.expandable_target'
+    },
+
+    /**
+     * Creates and customizes the Expandable.
+     *
+     * @param {object} options - Object used to customize the Expandable.
+     */
+    create: function create( options ) {
+      var properties = _extend( true, {}, Expandable.defaultProperties,
+        options );
+
+      // Using $.extend to create an instance of the Expandable.
+      // Extending dom elements is generally bad and should be
+      // avoided but we are doing so to maintain backwards compatibility.
+      // `this` is referring to the dom element.
+      _extend( this, Expandable, properties );
+      this.init();
+    },
+
+    /**
+     * Sets the initial state for the Expandable.
+     */
+    init: function init() {
+      this.isInAccordion = Boolean( this.$el.parents( '.expandable-group' )
+        .data( 'accordion' ) );
+      this.initUI();
+      this.initEvents();
+    },
+
+    /**
+     * Removes the dom element from the dom.
+     */
+    destroy: function destroy() {
+      if ( this.$el ) {
+        this.$el.remove();
+      }
+    },
 
-      var $this = $( this ),
-          $target = $this.find('.expandable_target').not( $this.find('.expandable .expandable_target') ),
-          $cueOpen = $this.find('.expandable_cue-open').not( $this.find('.expandable .expandable_cue-open') ),
-          $cueClose = $this.find('.expandable_cue-close').not( $this.find('.expandable .expandable_cue-close') ),
-          $content = $this.find('.expandable_content').not( $this.find('.expandable .expandable_content') ),
-          $groupParent = $this.parents('.expandable-group'),
-          accordion = $groupParent.length > 0 && $groupParent.data('accordion');
+    /**
+     * Initializes the initial state of the Expandable UI.
+     */
+    initUI: function initUI() {
+      var $el = this.$el;
 
-      if ( accordion ) {
-        var $siblings = $this.siblings('.expandable');
+      for ( var uiKey in this.ui ) {
+        if ( this.ui.hasOwnProperty( uiKey ) ) {
+          this['$' + uiKey] = $el.find( this.ui[uiKey] + ':first' );
+        }
       }
 
-      this.init = function() {
-        // Todo: recommend using an id on all expandables so that we can use
-        // the aria-controls attribute.
-        $target.attr( 'aria-controls', $content.attr('id') );
-        if ( $this.hasClass('expandable__expanded') ) {
-          this.expand( 0 );
-        } else {
-          this.collapse( 0 );
-        }
-        $target.on( 'click', $.proxy( this.handleClick, this ) );
-      };
-
-      this.handleClick = function( ev ) {
-        ev.preventDefault();
-        ev.stopPropagation();
-        this.toggle();
-        if ( accordion ) {
-          $siblings.each( function( index, sibling ) {
+      this.$target.attr( 'aria-controls', this.$content.attr( 'id' ) );
+      if ( $el.hasClass( this.expandedClass ) ) {
```

Comment:
I'd prefer if this state was stored outside of the DOM. Maybe by updating and pulling from `defaultProperties.isEcpanded`?